app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventViewModel.java

```diff
@@ -50,11 +58,33 @@
     @NonNull
     public abstract Boolean canBeEdited();
 
+    public boolean openedAttributeList = false;
+
+    public void toggleAttributeList() {
+        this.openedAttributeList = !this.openedAttributeList;
+    }
+
     @NonNull
     public static ProgramEventViewModel create(@NonNull String uid, @NonNull String orgUnitUid, @NonNull String orgUnitName, @NonNull Date date,
                                                @NonNull State eventState, @NonNull List<Pair<String, String>> data, @NonNull EventStatus status,
                                                @NonNull Boolean isExpired, @NonNull String attributeOptionComboName, Geometry geometry, Boolean canBeEdited) {
         return new AutoValue_ProgramEventViewModel(uid, orgUnitUid, orgUnitName, date, eventState, data, status, isExpired, attributeOptionComboName, geometry, canBeEdited);
     }
 
+    public SpannableStringBuilder getSpannableStringValues() {
+        SpannableStringBuilder stringBuilder = new SpannableStringBuilder();
+        for (Pair<String, String> nameValuePair : eventDisplayData()) {
+            if (!Objects.equals(nameValuePair.component2(), "-")) {
+                SpannableString value = new SpannableString(nameValuePair.component2());
+                int colorToUse = eventDisplayData().indexOf(nameValuePair) % 2 == 0 ? Color.parseColor("#8A333333") : Color.parseColor("#61333333");
+                value.setSpan(new ForegroundColorSpan(colorToUse), 0, value.length(), Spanned.SPAN_INCLUSIVE_INCLUSIVE);
+                stringBuilder.append(value);
+                if (eventDisplayData().indexOf(nameValuePair) != eventDisplayData().size() - 1) {
+                    stringBuilder.append(" ");
+                }
+            }
+        }
+        return stringBuilder;
```

Comment:
I miss the unit test from both view models :/ . Also, as @andresmr presentation logic should not be couple to android dependencies (you have color here)

Reply:
Yes, In my opinion, as I comented to @Balcan in a PR related to Strings, I would inject in the **Presenter** or **ViewModel** in this case, a **resourceProvider** with methods like `providePrimaryColor()` and so. Or more granular, different providers (contentProvider, colorProvider ...) as we want to do it.